### PR TITLE
Use .devices[] for modinst indexing

### LINKED ISSUE #852: nimodinst.Session is indexed directly rather than through repeated capability container
It appears we didn't introduce repeated capability container to nimodinst as part of #718. This is inconsistent. I suggest we turn `session[0].chassis_number` into `session.devices[0].chassis_number`.

It's more typing but it's also more consistent with the other nimi-python APIs.

### PR DESCRIPTION
**Pull request by texasaggie97**
- [X] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).
- [X] I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.
- [X] I've ~~added~~changed tests applicable for this pull request
### What does this Pull Request accomplish?
* Change indexing to .devices[] instead of on the session
  * `session[0].device_name` becomes `session.devices[0].device_name`
### List issues fixed by this Pull Request below, if any.
* Fix #852 

### What testing has been done?
* Unit
* System

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -98,6 +98,13 @@ All notable changes to this project will be documented in this file.
         * Methods removed
             * `get_frequency_response()` [#823](https://github.com/ni/nimi-python/issues/823)
             * `export_signal()` - [#828](https://github.com/ni/nimi-python/issues/828)
+* ### NI-ModInst
+    * #### Added
+    * #### Changed
+        * Indexing on the modinst session is no longer allowed
+            * `session[0].device_name` becomes `session.device[0].device_name`
```
Comment (marcoskirsch) [inline CHANGELOG.md]: should be `devices` (plural)

File: CHANGELOG.md [hunk 2]
```diff
@@ -98,6 +98,13 @@ All notable changes to this project will be documented in this file.
         * Methods removed
             * `get_frequency_response()` [#823](https://github.com/ni/nimi-python/issues/823)
             * `export_signal()` - [#828](https://github.com/ni/nimi-python/issues/828)
+* ### NI-ModInst
+    * #### Added
+    * #### Changed
+        * Indexing on the modinst session is no longer allowed
```
Comment (marcoskirsch) [inline CHANGELOG.md]: modinst session should be `nimodinst.Session` (proper Python class name, no ambiguity)

File: src/nidcpower/metadata/functions_addon.py [hunk 3]
```diff
@@ -6,7 +6,7 @@
 functions_codegen_method = {
     'InitializeWithChannels':          { 'codegen_method': 'private', 'method_name_for_documentation': '__init__', },
     'InitWithOptions':                 { 'codegen_method': 'no',       },
-    'Initiate':                        { 'codegen_method': 'private', 'public_method_name': 'initiate', },
+    'Initiate':                        { 'codegen_method': 'private', 'method_name_for_documentation': 'initiate', },
```
Comment (marcoskirsch) [inline src/nidcpower/metadata/functions_addon.py]: Should be called out in PR description, or moved to its own PR.

File: CHANGELOG.md (hunk 1 above)
Comment (texasaggie97) [reply CHANGELOG.md]: Changed

File: CHANGELOG.md (hunk 2 above)
Comment (texasaggie97) [reply CHANGELOG.md]: Changed

File: src/nidcpower/metadata/functions_addon.py (hunk 3 above)
Comment (texasaggie97) [reply src/nidcpower/metadata/functions_addon.py]: Made the change in master so that it didn't need to be in each branch
